+Base/GitPanel.m

```diff
@@ -0,0 +1,164 @@
+            
+            filedetails = ' &lt;file(s)&gt; (--all)';
+            commitdetails = ' --author &lt;you&gt; --message &lt;description&gt;';
+            
+            obj.menu = uicontextmenu(f);
+            uimenu(obj.menu, 'Label',  '<html><font color="purple">git</font> fetch', 'Callback', @(s,e)obj.update)
+            uimenu(obj.menu, 'Label', ['<html><font color="purple">git</font> status<br>'...
+                                             '<font color="purple">git</font> ad&d' filedetails '<br>'...
+                                             '<font color="purple">git</font> commit' commitdetails '<br>'...
+                                             '<font color="purple">git</font> push origin <branch>'], 'Callback', @(s,e)(disp('Add files to a commit, then commit, and push to save changes.')));
+            uimenu(obj.menu, 'Label', ['<html><font color="purple">git</font> status<br>',...
+                                             '<font color="purple">git</font> pull'], 'Callback', @(s,e)(disp('Pull to stay up to date')))
+            
+            obj.panel.UIContextMenu = obj.menu; 
+            
+            obj.control = uicontrol('Parent', obj.panel, 'Style', 'checkbox', 'UIContextMenu', obj.menu, 'Units', 'characters', 'String', '', 'Tag', 'gitpanelcontrol');     % Text does not display HTML :(
+            obj.control.Position(1:2) = [-2.75 0];
+            obj.control.Position(3) = 200;
+            
+            if (obj.enableLeftClick)
+                obj.panel.ButtonDownFcn = @(s,e)(obj.update);
+                obj.control.Callback = @(s,e)obj.update;
+            end
+            
+            obj.update();
+        end
+        function update(obj)
+            obj.panel.Children.Enable = 'off';
+            obj.panel.HighlightColor = [.5 0 .5];   % Purple
+            obj.panel.Title = 'git fetch';
+            drawnow;
+            
+            try
+                git('fetch -q --all');
+                
+                obj.panel.Children.Tooltip = obj.tooltip();
+                obj.panel.Children.String = obj.info();
+                
+                obj.panel.Children.Enable = 'on';
+                obj.panel.HighlightColor = 'w';
+                obj.panel.Title = 'git branch';
+                drawnow;
+            catch err
+                obj.panel.Children.Enable = 'on';
+                obj.panel.Children.String = '<html>Could not <font color="purple">git</font> fetch.';
+                obj.panel.Children.Tooltip = 'Something terrible happened.';
+                drawnow;
+                
+                rethrow(err);
+            end
+        end
+        function thisbranch = thisbranch(obj) %#ok<MANU> It wasn't fetching properly when it was static.
+            thisbranch = split(git('branch', '-v'), '* ');
+            thisbranch = split(thisbranch{end}, newline);
+            thisbranch = thisbranch{1};
+        end
+        function str = info(obj)
```

Comment:
So there are actually two methods that can be used to get git information: [git](https://github.com/mwalsh161/CommandCenter/blob/dev/HelperFunctions/ThirdParty/git.m) and [getGitInfo](https://github.com/mwalsh161/CommandCenter/blob/dev/HelperFunctions/ThirdParty/getGitInfo.m). `getGitInfo` returns a struct that might have some of this parsing already built in. If it doesn't have all that stuff, might be nice to build upon those helper functions to return nicer structs?

Reply:
I don't expect that people will use this outside of the panel. Plus, most of what I did was UI/HTML stuff to make it look nicer.